No binary available for node 16 (linux & mac)

**Describe the bug**
I face a strange issue with node 16. While trying to install dependencies npm can't find a binary of `dd-trace`. I didn't have such an issue on node 10. It might be related to node 16 or npm 7. Is it intentional? seems like binaries were available before. I have another binary (`node-sass`) - it works fine. 

```
npm ERR! code 1
npm ERR! path /builds/<reduced>/node_modules/dd-trace
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@16.2.0 | linux | x64
npm ERR! gyp ERR! find Python 
npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
npm ERR! gyp ERR! find Python checking if "python3" can be used
npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error
npm ERR! gyp ERR! find Python checking if "python" can be used
npm ERR! gyp ERR! find Python - "python" is not in PATH or produced an error
npm ERR! gyp ERR! find Python checking if "python2" can be used
npm ERR! gyp ERR! find Python - "python2" is not in PATH or produced an error
npm ERR! gyp ERR! find Python 
npm ERR! gyp ERR! find Python **********************************************************
npm ERR! gyp ERR! find Python You need to install the latest version of Python.
npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
npm ERR! gyp ERR! find Python you can try one of the following options:
npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)
npm ERR! gyp ERR! find Python - Set the environment variable PYTHON
npm ERR! gyp ERR! find Python - Set the npm configuration variable python:
npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
npm ERR! gyp ERR! find Python For more information consult the documentation at:
npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
npm ERR! gyp ERR! find Python **********************************************************
npm ERR! gyp ERR! find Python 
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
npm ERR! gyp ERR! stack     at PythonFinder.fail (/builds/<reduced>/node_modules/node-gyp/lib/find-python.js:302:47)
npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/builds/<reduced>/node_modules/node-gyp/lib/find-python.js:136:21)
npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/builds/<reduced>/node_modules/node-gyp/lib/find-python.js:179:16)
npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/builds/<reduced>/node_modules/node-gyp/lib/find-python.js:266:16)
npm ERR! gyp ERR! stack     at exithandler (node:child_process:333:5)
npm ERR! gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:345:5)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
npm ERR! gyp ERR! stack     at onErrorNT (node:internal/child_process:480:16)
npm ERR! gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
npm ERR! gyp ERR! System Linux 4.19.78-coreos
npm ERR! gyp ERR! command "/usr/local/bin/node" "/builds/<reduced>/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /builds/<reduced>/node_modules/dd-trace
npm ERR! gyp ERR! node -v v16.2.0
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-05-28T14_32_17_165Z-debug.log
```

Dockerfile
```dockerfile
FROM node:16-slim as base
ENV NODE_ENV=production
# Make npm binaries available directly in terminal (e.g. ts-node-dev, npm-run-all, eslint, etc.)
ENV PATH=/app/node_modules/.bin:$PATH
# Install Heroku CLI (curl is needed only to fetch heroku intaller)
RUN apt-get update \
  && apt-get install -y curl ca-certificates --no-install-recommends \
  && rm -rf /var/lib/apt/lists/* \
  && curl https://cli-assets.heroku.com/install.sh | sh \
  && apt-mark auto '.*' > /dev/null \
  && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false
# By default, Docker runs container as root which inside of the container can pose as a security issue.
# That's why it's better to run everything as node and keep correct permissions (node:node)
RUN mkdir /app && chown -R node:node /app
WORKDIR /app
USER node
# This layer (base) will be used from cache unless package.json or package-lock.json is changed
COPY --chown=node:node package.json package-lock.json ./
# `npm ci` is more reliable and quicker
RUN npm ci --quiet && npm cache clean --force

FROM base as dev
ENV NODE_ENV=development
# prevent cypress binary installation inside docker, we usually run it on host machine
ENV CYPRESS_INSTALL_BINARY=0
# Only dev deps are needed since prod deps are already installed in the `base` layer
RUN npm install --quiet --no-optional --no-audit --no-fund && npm cache clean --force
CMD ["npm", "run", "heroku"]
```

package.json (reduced)
```json
{
  "name": "test",
  "private": true,
  "version": "1.0.0",
  "author": "",
  "dependencies": {
    "connect-datadog": "0.0.9",
    "dd-trace": "^0.34.0",
    "express": "^4.17.1"
  },
  "engines": {
    "node": "16.x"
  }
}

```
**Environment** 

* **Operation system:** Linux (in docker)/ Mac
* **Node version:** 16
* **Tracer version:** Latest (0.34.0)
* **Agent version:** 0.0.9

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1597,6 +1597,8 @@ workflows:
       - alpine-ia32:
           <<: *prebuild-job
       # Darwin x64
+      - mac-x64-16:
+          <<: *prebuild-job
       - mac-x64-14:
           <<: *prebuild-job
       - mac-x64-12:
@@ -1606,6 +1608,8 @@ workflows:
       - mac-x64-8:
           <<: *prebuild-job
       # Darwin ia32
+      - mac-ia32-16:
+          <<: *prebuild-job
       - mac-ia32-14:
           <<: *prebuild-job
       - mac-ia32-12:
@@ -1615,6 +1619,8 @@ workflows:
       - mac-ia32-8:
           <<: *prebuild-job
       # Windows x64
+      - win-x64-16:
+          <<: *prebuild-job
       - win-x64-14:
           <<: *prebuild-job
       - win-x64-12:
@@ -1624,6 +1630,8 @@ workflows:
       - win-x64-8:
           <<: *prebuild-job
       # Windows ia32
+      - win-ia32-16:
+          <<: *prebuild-job
       - win-ia32-14:
           <<: *prebuild-job
       - win-ia32-12:
@@ -1646,18 +1654,22 @@ workflows:
             - linux-ia32-8-test
             - alpine-x64
             - alpine-ia32
+            - mac-x64-16
             - mac-x64-14
             - mac-x64-12
             - mac-x64-10
             - mac-x64-8
+            - mac-ia32-16
             - mac-ia32-14
             - mac-ia32-12
             - mac-ia32-10
             - mac-ia32-8
+            - win-x64-16
             - win-x64-14
             - win-x64-12
             - win-x64-10
             - win-x64-8
+            - win-ia32-16
             - win-ia32-14
             - win-ia32-12
             - win-ia32-10

